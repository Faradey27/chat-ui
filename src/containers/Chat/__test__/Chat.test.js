
import ChatDriver from './Chat.driver';

describe('Chat', () => {
  let driver = null;

  beforeEach(() => {
    driver = new ChatDriver();
  });

  it('renders without crashing', () => {
    expect(driver.when.render().is.ok()).toBeTruthy();
  });

  it('should connect socket on mount', () => {
    expect(driver.when.render().is.socketConnected()).toBeTruthy();
  });

  it('should set username, useravatar, user message and send it on click', () => {
    const addMessage = jest.fn();
    driver
      .when.render({addMessage})
      .when.nameEntered('name')
      .when.messageEntered('message')
      .when.sendMessage();
    expect(driver.get.emitMockedFunc().mock.calls.length).toBe(1);
    expect(driver.get.emitMockedFunc().mock.calls[0][1].text).toBe('message');
    expect(driver.get.emitMockedFunc().mock.calls[0][1].user.name).toBe('name');
  });
});

