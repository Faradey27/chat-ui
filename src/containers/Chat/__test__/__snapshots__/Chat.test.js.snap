// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat snapshot test 1`] = `
<div
  className="css-1cv63l5"
  data-hook="chat"
>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#fafafa",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100,
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": -16,
          "marginRight": 8,
          "marginTop": 8,
          "muiPrepared": true,
        }
      }
    >
      <span />
    </div>
    <h1
      onClick={[Function]}
      style={
        Object {
          "boxFlex": 1,
          "color": "initial",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      Chat UI
    </h1>
  </div>
  <div
    className="css-1d9v50n"
  >
    <div
      className="css-1yyd33c"
      data-hook="message-list-area"
    >
      <div
        className="css-12e3dyv"
      >
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <p>
            No messages
          </p>
          <p
            className="css-6btlr7"
          >
            Try to type and send something
          </p>
        </div>
      </div>
    </div>
    <div
      className="css-1tz2cim"
      data-hook="message-creation-area"
    >
      <div
        className="css-1p0g30k"
        data-hook="username-text-field"
      >
        <div
          data-hook="avatar"
        >
          <img
            className={undefined}
            size={40}
            src="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "muiPrepared": true,
                "userSelect": "none",
                "width": 40,
              }
            }
          />
        </div>
        <input
          className="css-nk00bm"
          data-hook="username-text-field-input"
          onChange={[Function]}
          value=""
        />
      </div>
      <form
        className="css-1xm32e0"
        onSubmit={[Function]}
      >
        <div
          className="css-ov1ktg"
          data-hook="message-text-field"
        >
          <input
            className="css-2kicle"
            data-hook="message-text-field-textarea"
            onChange={[Function]}
            placeholder="Type something ..."
            value=""
          />
        </div>
        <div
          data-hook="send-button"
        >
          <button
            className="css-11xn1r8"
            data-hook="send-button-el"
            onClick={undefined}
            type="submit"
          >
            <svg
              height="24"
              viewBox="0 0 48 48"
              width="24"
            >
              <path
                d="M4.02 42l41.98-18-41.98-18-.02 14 30 4-30 4z"
              />
              <path
                d="M0 0h48v48h-48z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
